### PR TITLE
Issue #PS-1550 feat: API integration to schedule offline events

### PR DESCRIPTION
**Pull request by suvarnakale**

### CODE REVIEW

File: src/components/PlannedSession.tsx [hunk 1]
```diff
@@ -217,19 +248,29 @@ const PlannedSession: React.FC<PlannedModalProps> = ({
...
     const startDatetime = convertToUTC(combinedStartDateTime);
     const endDatetime = convertToUTC(combinedEndDateTime);
-
-    if (startDatetime && endDatetime) {
+    const endDateValue = convertToUTC(combinedEndDateValue);
+
+    if (startDatetime && endDatetime && endDateValue) {
+      let isRecurringEvent: boolean;
+      if (startDate !== endDate) {
+        isRecurringEvent = true;
```
Comment (itsvick) [inline src/components/PlannedSession.tsx:261]: @suvarnakale try using ternary operator wherever possible

File: src/components/PlannedSession.tsx [hunk 2]
```diff
@@ -274,27 +371,177 @@ const PlannedSession: React.FC<PlannedModalProps> = ({
...
+      // Fetch cohort members if cohortId is available
+      if (cohortId) {
+        const filters = {
+          cohortId,
+          role: Role.STUDENT,
+          status: [Status.ACTIVE],
+        };
+
+        const response = await getMyCohortMemberList({
+          limit: 300,
```
Comment (itsvick) [inline src/components/PlannedSession.tsx:419]: @suvarnakale it should not be 300 now try to use limit and offset

File: src/components/PlannedSession.tsx [hunk 3]
```diff
@@ -274,27 +371,177 @@ const PlannedSession: React.FC<PlannedModalProps> = ({
...
+          attendeeCount = resp.totalCount || 0;
+          attendeeArray =
+            resp.userDetails?.map((attendee: any) => attendee.userId) || [];
+        }
+      }
+
+      // Determine title based on clickedBox and mode
+      let title = '';
+      if (clickedBox === 'PLANNED_SESSION') {
+        title = mode === 'online' ? 'Recurring Online' : 'Recurring Offline';
```
Comment (itsvick) [inline src/components/PlannedSession.tsx:436]: @suvarnakale translate

File: src/components/PlannedSession.tsx [hunk 4]
```diff
@@ -274,27 +371,177 @@ const PlannedSession: React.FC<PlannedModalProps> = ({
...
+          autoEnroll: true,
+          location: cohortName || '',
+          // longitude: '',
+          // latitude: '',
+          maxAttendees: attendeeCount,
+          attendees: attendeeArray,
+          status: 'live',
+          createdBy: userId,
+          updatedBy: userId,
+          idealTime: '120',
```
Comment (itsvick) [inline src/components/PlannedSession.tsx:458]: @suvarnakale what is this time? add it in app.constant with appropriate variable name

File: src/components/PlannedSession.tsx [hunk 5]
```diff
@@ -274,27 +371,177 @@ const PlannedSession: React.FC<PlannedModalProps> = ({
...
+          isRecurring: block?.isRecurring || false,
+          startDatetime: block?.startDatetime || '',
+          endDatetime: block?.endDatetime || '',
+          registrationStartDate: '',
+          registrationEndDate: '',
+          recurrencePattern: {
+            frequency:
+              block?.selectedWeekDays?.length === 7 ? 'daily' : 'weekly',
```
Comment (itsvick) [inline src/components/PlannedSession.tsx:466]: @suvarnakale add 7 in the app.constant

File: src/components/PlannedSession.tsx [hunk 6]
```diff
@@ -274,27 +371,177 @@ const PlannedSession: React.FC<PlannedModalProps> = ({
...
+        };
+
+        // Add meetingDetails only if sessionMode is 'online'
+        if (block?.sessionMode === 'online') {
+          (baseBody.onlineProvider = block?.onlineProvider || ''),
+            (baseBody.isMeetingNew = false),
+            (baseBody.meetingDetails = {
+              url: block?.meetingLink || '',
+              password: block?.meetingPasscode || '7674534',
+              id: '123-456-789',
```
Comment (itsvick) [inline src/components/PlannedSession.tsx:499]: @suvarnakale why meeting passcode and id hard coded

Comment (suvarnakale) [reply src/components/PlannedSession.tsx:499]: Bcoz those fields should be optional and for now from api side those are mandotory, if I didn't pass api will fail. so just for now kept it hardcoded.

File: src/components/PlannedSession.tsx (hunk 5 above)
Comment (suvarnakale) [reply src/components/PlannedSession.tsx:466]: okay

File: src/components/PlannedSession.tsx (hunk 4 above)
Comment (suvarnakale) [reply src/components/PlannedSession.tsx:458]: okay

File: src/components/PlannedSession.tsx (hunk 3 above)
Comment (suvarnakale) [reply src/components/PlannedSession.tsx:436]: okay

**Pull request merged**